pkg/client/interfaces.go

```diff
@@ -73,11 +73,22 @@ type RetrieveConfig struct {
 }
 
 // SonobuoyClient is a high-level interface to Sonobuoy operations.
-type SonobuoyClient struct{}
+type SonobuoyClient struct {
+	RestConfig *rest.Config
+	Client     kubernetes.Interface
+}
 
 // NewSonobuoyClient creates a new SonobuoyClient
-func NewSonobuoyClient() *SonobuoyClient {
-	return &SonobuoyClient{}
+func NewSonobuoyClient(restConfig *rest.Config) (*SonobuoyClient, error) {
+	sc := &SonobuoyClient{
+		RestConfig: restConfig,
+	}
+	clientset, err := kubernetes.NewForConfig(sc.RestConfig)
```

Comment:
What about the cases where we don't actually need a kube config? `sonobuoy gen` shouldn't require it

Reply:
I think it's weird that creating the dynamic client fails at call time, but creating a kubectl client fails at creation time

Reply:
> What about the cases where we don't actually need a kube config? sonobuoy gen shouldn't require it 

Gen should really be doing dynamic RBAC detection, so although it's not strictly needed right this instant, I think it's a better practice. 

> I think it's weird that creating the dynamic client fails at call time, but creating a kubectl client fails at creation time 

We could cache the dynamic client here too.  I don't have strong feeling on it.

Reply:
I think it's good for `sonobuoy gen` to not hit the network. It's a bit like `GET` requests having side effects, I think it would be surprising.

I'm happy to cache a dynamic client as well, my point was just we're initialising a struct only half our commands end up using.

Reply:
Right now gen doesn't actually hit the network, but the whole client having a valid KubeConfig in order to function seems like a reasonable contract.  It's not strictly needed by gen (again right now), but I don't want to open up a loophole in the api to allow for inconsistency.  

I'll add a //TODO to cache the dynamic client.